basyx.submodelservice-core/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/value/mapper/AnnotatedRelationshipElementValueMapper.java

```diff
@@ -0,0 +1,89 @@
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+package org.eclipse.digitaltwin.basyx.submodelservice.value.mapper;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.digitaltwin.aas4j.v3.model.AnnotatedRelationshipElement;
+import org.eclipse.digitaltwin.aas4j.v3.model.DataElement;
+import org.eclipse.digitaltwin.aas4j.v3.model.Reference;
+import org.eclipse.digitaltwin.basyx.submodelservice.value.AnnotatedRelationshipElementValue;
+import org.eclipse.digitaltwin.basyx.submodelservice.value.ReferenceValue;
+import org.eclipse.digitaltwin.basyx.submodelservice.value.ValueOnly;
+import org.eclipse.digitaltwin.basyx.submodelservice.value.factory.SubmodelElementValueMapperFactory;
+
+/**
+ * Maps {@link AnnotatedRelationshipElement} value to {@link AnnotatedRelationshipElementValue}
+ * 
+ * @author danish
+ *
+ */
+public class AnnotatedRelationshipElementValueMapper implements ValueMapper<AnnotatedRelationshipElementValue>{
+	private AnnotatedRelationshipElement annotatedRelationshipElement;
+
+	public AnnotatedRelationshipElementValueMapper(AnnotatedRelationshipElement relationshipElement) {
+		this.annotatedRelationshipElement = relationshipElement;
+	}
+
+	@Override
+	public AnnotatedRelationshipElementValue getValue() {
+		return new AnnotatedRelationshipElementValue(createReferenceValue(annotatedRelationshipElement.getFirst()),
+				createReferenceValue(annotatedRelationshipElement.getSecond()),
+				createValueOnly(annotatedRelationshipElement.getAnnotations()));
+	}
+
+	@Override
+	public void setValue(AnnotatedRelationshipElementValue annotatedRelationshipElementValue) {
+		setReferenceValue(annotatedRelationshipElement.getFirst(), annotatedRelationshipElementValue.getFirst());
+		setReferenceValue(annotatedRelationshipElement.getSecond(), annotatedRelationshipElementValue.getSecond());
+		setDataElementValue(annotatedRelationshipElement.getAnnotations(), annotatedRelationshipElementValue.getAnnotation());
+	}
+	
+	private void setDataElementValue(List<DataElement> annotations, List<ValueOnly> valueOnlies) {
+		annotations.stream().forEach(annotation -> setValue(annotation, valueOnlies));
+	}
+	
+	@SuppressWarnings("unchecked")
+	private void setValue(DataElement dataElement, List<ValueOnly> valueOnlies) {
+		@SuppressWarnings("rawtypes")
```

Comment:
Please either use the generic type of ValueMapper - or use <?> if there is no other possibility.